app/routes/shop.py

```diff
@@ -146,4 +148,54 @@ def upload(id):
   setattr(dive_shop, 'logo_img', s3_url)
   db.session.commit()
 
-  return { 'msg': 'dive shop successfully updated' }
\ No newline at end of file
+  return { 'msg': 'dive shop successfully updated' }
+
+@bp.route('/typeahead')
+@cache.cached(query_string=True)
+def get_typeahead():
+  query = request.args.get('query')
+  results = []
+  dive_shops_by_name = DiveShop.query \
+    .filter(DiveShop.name.ilike('%'+query+'%')) \
+    .limit(25) \
+    .all()
+  
+  dive_shops_by_city = DiveShop.query \
+    .filter(DiveShop.city.ilike('%'+query+'%')) \
+    .limit(25) \
+    .all()
+
+  dive_shops_by_state = DiveShop.query \
+    .filter(DiveShop.state.ilike('%'+query+'%')) \
+    .limit(25) \
+    .all()
+
+  for shop in dive_shops_by_name:
+    result = {
+      'id': shop.id,
+      "text": shop.name,
+      'subtext': shop.city
```

Comment:
can we do `f'{shop.city}, {shop.state}'`

Reply:
Sure.